File: guard/t.go

package main

import (
       "time"
       "context"
       "fmt"
       "encoding/json"
       "github.com/algorand/go-algorand-sdk/client/v2/indexer"
       "github.com/algorand/go-algorand-sdk/client/v2/common/models"
)

const indexerAddress = "http://localhost:8980"
const indexerToken = ""

func lookAtTxn(t models.Transaction) {
  var at = t.ApplicationTransaction
  if len(at.ApplicationArgs) == 0 {
    return
  }

  JSON, err := json.Marshal(t)
  _ = err
  fmt.Printf(string(JSON))

  fmt.Printf("%d\n", at.ApplicationId)
  if string(at.ApplicationArgs[0]) == "publishMessage" { // The note filter is effectively the same thing
      var vaa = at.ApplicationArgs[1]
      fmt.Printf(t.Sender + " -> " + string(vaa) + "\n")
  }
}

func main() {
     indexerClient, err := indexer.MakeClient(indexerAddress, indexerToken)
     _ = err

     // Parameters
     var notePrefix = "publishMessage"
     var next_round uint64 = 0

     for true {
         var nextToken = ""
         for true {
             result, err := indexerClient.SearchForTransactions().NotePrefix([]byte(notePrefix)).MinRound(next_round).NextToken(nextToken).Do(context.Background())
             _ = err

             for i := 0; i < len(result.Transactions); i++ {
                var t = result.Transactions[i]
                if len(t.InnerTxns) > 0 {
                    for q := 0; q < len(t.InnerTxns); q++ {
                      lookAtTxn(t.InnerTxns[q])
                    }
                } else {
                  lookAtTxn(t)
                }
           }

             if result.NextToken != "" {
                 nextToken = result.NextToken
             } else {
                 next_round = result.CurrentRound + 1
                 break
             }
         }
         time.Sleep(1 * time.Second)
     }
}

File: guard/t.py
import os
import json
from time import sleep
from datetime import timezone
import pprint
import base64
import time
import sys

from algosdk.v2client import indexer

indexer_token = "a" * 64
myindexer = indexer.IndexerClient(indexer_token=indexer_token, indexer_address='http://localhost:8980')

note_prefix = 'publishMessage'.encode()

next_round = 24

while True:
    nexttoken = ""
    while True:
        response = myindexer.search_transactions( min_round=next_round, note_prefix=note_prefix, next_page=nexttoken)
        pprint.pprint(response)
        if 'next-token' in response:
            nexttoken = response['next-token']
        else:
            next_round = response['current-round'] + 1
            break
    time.sleep(1)
